modules/ghlint-rules/src/main/kotlin/net/twisterrob/ghlint/rules/NeverExpressionRule.kt

```diff
@@ -0,0 +1,72 @@
+package net.twisterrob.ghlint.rules
+
+import net.twisterrob.ghlint.model.WorkflowStep
+import net.twisterrob.ghlint.rule.Example
+import net.twisterrob.ghlint.rule.Issue
+import net.twisterrob.ghlint.rule.Reporting
+import net.twisterrob.ghlint.rule.report
+import net.twisterrob.ghlint.rule.visitor.VisitorRule
+import net.twisterrob.ghlint.rule.visitor.WorkflowVisitor
+
+public class NeverExpressionRule : VisitorRule, WorkflowVisitor {
+
+    override val issues : List<Issue> = listOf(NeverExpression)
+
+    private companion object {
+        const val uses = "actions/checkout@\${{ github.sha }}"
+
+        val NeverExpression = Issue(
+            id = "NeverExpression",
```

Comment:
What do you think of `InvalidExpressionUsage`? I agree - the "Never" to me implies that it's a strong opinion, rather than something that actually just doesn't work.